specs/release.md

```diff
@@ -0,0 +1,319 @@
+
+### GitHub Deployments API Integration
+
+All deployments (development and production) MUST create GitHub Deployment records via the Deployments API. This provides centralized deployment history, status tracking, and integration with GitHub's environment protection features.
+
+For each deployment, the workflow MUST:
+
+1. Create a deployment record with the commit SHA being deployed
+2. Set the environment name ("development" or "production")
+3. Set deployment status to "in_progress" during deployment
+4. Update status to "success" if deployment and verification succeed
+5. Update status to "failure" if any step fails
+6. Include deployment URL in the status update
+
+GitHub Environment protection rules SHOULD be configured for production to:
+
+- Require manual approval from designated reviewers
+- Enforce deployment branch restrictions (only `main`)
+- Apply wait timer (optional, e.g., 5 minute cooldown between deployments)
+
+## Versioning Strategy
+
+The project DOES NOT use semantic versioning tags for deployments. Version numbers exist for documentation and tracking purposes but do NOT trigger deployments or control which code runs in production.
+
+Version numbers in `package.json` SHOULD be updated manually when significant features or breaking changes are released, following semantic versioning principles (MAJOR.MINOR.PATCH). These version updates help users understand release scope but are decoupled from the deployment process.
+
+Git tags MAY be created for significant milestones or release markers, but tags MUST NOT trigger automated deployments. Tags serve as bookmarks in history, not deployment triggers.
+
+## Deployment Verification
+
+Every deployment MUST be verified before being considered successful. Verification failures MUST trigger automatic rollback to the previous stable deployment.
+
+### Health Check Verification
+
+Immediately after deployment, the workflow MUST:
+
+1. Wait 30 seconds for edge network propagation
+2. Request the `/health` endpoint of the deployed Worker
+3. Verify HTTP 200 response with valid JSON health status
+4. Fail the deployment if health check does not pass within 3 attempts
+
+The health check endpoint is defined in the Deployment Specification and MUST NOT require authentication.
+
+### Smoke Test Verification
+
+Production deployments SHOULD execute smoke tests against the newly deployed environment. Smoke tests MUST:
+
+- Validate OAuth client registration succeeds
+- Verify MCP initialize endpoint returns valid session
+- Confirm at least one tool execution succeeds (e.g., `glob` on bootstrap files)
+- Complete within 60 seconds total
+
+Smoke tests MAY be temporarily disabled during CI/CD infrastructure changes, documented with TODO comments and GitHub issues tracking re-enablement.
+
+## Rollback Procedures
+
+When a deployment fails verification or is identified as problematic post-deployment, the system MUST support rapid rollback to the previous known-good deployment.
```

Comment:
Need to think about how we manage breaking changes caused by things like data migrations. Ideally we would avoid walking through one way doors and always follow an expand and contract methodology.